### PR TITLE
Remove sigmoid in `BinaryPrecisionRecallCurve`

### PR DESCRIPTION
**Pull request by PussyCat0700**
# What does this PR do?
This PR addresses [#3179](https://github.com/Lightning-AI/torchmetrics/issues/3179) by making the use of sigmoid normalization optional in `BinaryPrecisionRecallCurve`. Specifically, it introduces a normalization parameter to _binary_precision_recall_curve_format and disables automatic sigmoid application in `BinaryPrecisionRecallCurve`.

# Changes
- Adds a normalization: `Optional[str] = "sigmoid"` argument to _binary_precision_recall_curve_format, allowing users to override the default behavior.

- Passes normalization=None in `BinaryPrecisionRecallCurve` to preserve raw logits and prevent batch-wise distortion during `.update()` calls.

# Notes
- This fix only applies to `BinaryPrecisionRecallCurve`, which is the class I directly encountered issues with.

- I’ve noticed that other metric classes may also apply sigmoid implicitly in ways that could cause similar inconsistencies across batch-wise updates, but since I have not tested those cases, I am leaving them unchanged to avoid unintended side effects.

----
📚 Documentation preview 📚: https://torchmetrics--3182.org.readthedocs.build/en/3182/

### CODE REVIEW

**Comment by Borda** [comment]
@PussyCat0700 pls check the failing tests

**Comment by PussyCat0700** [comment]
This commit should pass all the checks, as I default normalization back to `sigmoid` for compatibility of unittests. This is done by adding a `normalization` parameter in `BinaryPrecisionRecallCurve`. Users wishing to accept negative logits can simply set it to False.

File: CHANGELOG.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
--
+- Fixed unintended `sigmoid` normalization in `BinaryPrecisionRecallCurve`, which might cause incorrect ROC computation when updating incrementally with raw logits. ([#3179](https://github.com/Lightning-AI/torchmetrics/pull/3179)). This fix will only take effect when setting `disable_softmax` to False (defaults to True for compatibility).
```
Comment (Borda) [inline CHANGELOG.md]:
```suggestion
- Fixed unintended `sigmoid` normalization in `BinaryPrecisionRecallCurve` ([#3182](https://github.com/Lightning-AI/torchmetrics/pull/3182))
```

File: src/torchmetrics/classification/precision_recall_curve.py [hunk 2]
```diff
@@ -139,6 +142,7 @@ def __init__(
         thresholds: Optional[Union[int, list[float], Tensor]] = None,
         ignore_index: Optional[int] = None,
         validate_args: bool = True,
+        normalization: Optional[str] = "sigmoid",
```
Comment (Borda) [inline src/torchmetrics/classification/precision_recall_curve.py]:
```suggestion
        normalization: Optional[Literal["sigmoid", "softmax"]] = "sigmoid",
```
